src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,100 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elementData;
+    private int size;
+
+    public ArrayList() {
+        this.elementData = new Object[DEFAULT_CAPACITY];
+    }
+
+    private void checkIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException("ArrayList index out of bounds");
```

Comment:
add `index `and `size `to the message
https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/array-list.html#make-your-exceptions-informative
![image](https://user-images.githubusercontent.com/20326858/216031969-2f029281-3a43-4629-b526-c687991dceba.png)